samza-sql/src/main/java/org/apache/samza/sql/interfaces/SqlIOConfig.java

```diff
@@ -96,10 +97,15 @@ public SqlIOConfig(String systemName, String streamName, List<String> sourcePart
     streamConfigs.remove(CFG_SAMZA_REL_CONVERTER);
     streamConfigs.remove(CFG_REL_SCHEMA_PROVIDER);
 
-    // Currently, only local table is supported. And it is assumed that all tables are local tables.
-    if (tableDescriptor != null) {
-      streamConfigs.put(String.format(StreamConfig.BOOTSTRAP_FOR_STREAM_ID(), streamName), "true");
-      streamConfigs.put(String.format(StreamConfig.CONSUMER_OFFSET_DEFAULT_FOR_STREAM_ID(), streamName), "oldest");
+    if (!isRemoteTable()) {
+      // The below config is required for local table and streams but not for remote table.
+      streamConfigs.put(String.format(StreamConfig.PHYSICAL_NAME_FOR_STREAM_ID(), streamId), streamName);
+    }
+
+    if (tableDescriptor != null && !isRemoteTable()) {
```

Comment:
Can you merge this condition with the previous one?

Reply:
Yup. Thanks for the suggestion.